### PR TITLE
Initial version of E2E Test Cases and setSeed address loading issue

### PR DESCRIPTION
**Pull request by erdimaden**
### What changed? Why?
- Adding initial version of e2e test cases.
- Adding NPM dry-run to e2e workflow.
- e2e tests are now part of PR workflow.
- Fixing setSeed address loading issue.

### CODE REVIEW

File: src/coinbase/tests/e2e.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    await unhydratedWallet.loadSeed("test_seed.json");
+    expect(unhydratedWallet.canSign()).toBe(true);
+    expect(unhydratedWallet.getId()).toBe(walletId);
+
+    console.log("Transfering 1 Gwei from default address to second address...");
+    const transfer = await unhydratedWallet.createTransfer(1, Coinbase.assets.Gwei, wallet);
+    expect(await transfer.getStatus()).toBe(TransferStatus.COMPLETE);
+    console.log(`Transferred 1 Gwei from ${unhydratedWallet} to ${wallet}`);
+
+    console.log("Fetching updated balances...");
```
Comment (jazz-cb) [inline src/coinbase/tests/e2e.ts:83]: can we add one more transfer request that transfers funds back to the CB faucet `0x6Cd01c0F55ce9E0Bf78f5E90f72b4345b16d515d` ?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -19,3 +19,14 @@ jobs:
 
     - name: Run Unit Tests
       run: npm run test
+
+    - name: Run E2E Tests
```
Comment (jazz-cb) [inline .github/workflows/test.yml:23]: could you add instructions somewhere to run this locally as well?

Comment (erdimaden) [reply .github/workflows/test.yml:23]: Sure, I added npm scripts to we can run it locally and README is also updated.

File: src/coinbase/tests/e2e.ts (hunk 1 above)
Comment (erdimaden) [reply src/coinbase/tests/e2e.ts:93]: Synced on this but the faucet contract is updated. we will have a followup

**Review by jazz-cb — APPROVED** [review]

File: src/coinbase/tests/e2e.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    console.log("Creating new wallet...");
+    const wallet = await user.createWallet();
+    expect(wallet?.getId()).toBeDefined();
+    console.log(
+      `Created new wallet with ID: ${wallet.getId()}, default address: ${wallet.getDefaultAddress()}`,
+    );
+
+    console.log("Importing wallet with balance...");
+    const seedFile = JSON.parse(process.env.SEED || "");
```
Comment (jazz-cb) [inline src/coinbase/tests/e2e.ts:45]: this should be process.env.SEED_FILE instead of process.env.SEED. It is confusing

Comment (erdimaden) [reply src/coinbase/tests/e2e.ts:45]: @jazz-cb sure, it is content of the seed file. Maybe SEED_FILE_CONTENT?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -19,3 +19,12 @@ jobs:
...
```
Comment (alex-stone) [inline .github/workflows/test.yml:23]: Should we run the E2E tests in their own github action separate from unit tests?

Will help in reporting in identifying what is the error more quickly

Comment (erdimaden) [reply .github/workflows/test.yml]: makes sense, I created a workflow file for e2e tests.

**Review by jazz-cb — APPROVED** [review]

**Review by jazz-cb — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -165,16 +165,16 @@ In order to persist the data for the Wallet, you will need to implement a store
 await store(data);
 ```
 
-For convenience during testing, we provide a `saveWallet` method that stores the Wallet data in your local file system. This is an insecure method of storing wallet seeds and should only be used for development purposes.
+For convenience during testing, we provide a `saveSeed` method that stores the Wallet data in your local file system. This is an insecure method of storing wallet seeds and should only be used for development purposes.
 
 ```typescript
-user.saveWallet(wallet);
+user.saveSeed(wallet);
```
Comment (jazz-cb) [inline README.md:171]: wallet.saveSeed

File: README.md [hunk 6]
```diff
@@ -165,16 +165,16 @@ In order to persist the data for the Wallet, you will need to implement a store
 await store(data);
 ```
 
-For convenience during testing, we provide a `saveWallet` method that stores the Wallet data in your local file system. This is an insecure method of storing wallet seeds and should only be used for development purposes.
+For convenience during testing, we provide a `saveSeed` method that stores the Wallet data in your local file system. This is an insecure method of storing wallet seeds and should only be used for development purposes.
```
Comment (jazz-cb) [inline README.md:168]: `saveSeed` method that stores the wallet's seed in your local file system

File: README.md [hunk 7]
```diff
@@ -165,16 +165,16 @@ In order to persist the data for the Wallet, you will need to implement a store
...
 
 ```typescript
-user.saveWallet(wallet);
+wallet.saveSeed(wallet);
 ```
 
-To encrypt the saved data, set encrypt to true. Note that your CDP API key also serves as the encryption key for the data persisted locally. To re-instantiate wallets with encrypted data, ensure that your SDK is configured with the same API key when invoking `saveWallet` and `loadWallets`.
+To encrypt the saved data, set encrypt to true. Note that your CDP API key also serves as the encryption key for the data persisted locally. To re-instantiate wallets with encrypted data, ensure that your SDK is configured with the same API key when invoking `saveSeed` and `loadWallets`.
```
Comment (jazz-cb) [inline README.md:174]: `saveSeed` and `loadSeed`.

**Review by jazz-cb — APPROVED** [review]

**Pull request merged**